share/initialization/common

```diff
@@ -182,7 +182,23 @@ fi
 : ${ECHO_STYLE:=RAW}
 
 # variables needed for command history
-HISTFILE=~/.yash_history HISTSIZE=5000
+
+HISTFILE=${XDG_STATE_HOME:-~/.local/state}/yash/history
+HISTSIZE=5000
+
+
+# handle old yash history directory
+if [ -f "$HOME/.yash_history" ] && [ -z "$HISTFILE" ]; then
```

Comment:
This conditional is faulty now since `HISTFILE` is set on the lines above. Do you have any suggestions on a more elegant way to check if the user is using the older path other than just checking if the file exists?

Perhaps instead of trying to set the old `HISTFILE`, we could just give a warning to the user stating they will need to set `HISTFILE`.